core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1321,7 +1321,113 @@ private AcquisitionLockTimerTask acquisitionLockTimerTask(
     }
 
     private void releaseAcquisitionLockOnTimeout(String memberId, long firstOffset, long lastOffset) {
-        // TODO: Implement the logic to release the acquisition lock on timeout.
+        lock.writeLock().lock();
+        try {
+            Map.Entry<Long, InFlightBatch> floorOffset = cachedState.floorEntry(firstOffset);
+            if (floorOffset == null) {
+                log.error("Base Offset {} not found for share partition: {}-{}", firstOffset, groupId, topicIdPartition);
+                return;
+            }
+            List<PersisterStateBatch> stateBatches = new ArrayList<>();
+            NavigableMap<Long, InFlightBatch> subMap = cachedState.subMap(floorOffset.getKey(), true, lastOffset, true);
+            for (Map.Entry<Long, InFlightBatch> entry : subMap.entrySet()) {
+                InFlightBatch inFlightBatch = entry.getValue();
+
+                if (inFlightBatch.offsetState() == null
+                        && inFlightBatch.batchState() == RecordState.ACQUIRED
+                        && checkForStartOffsetWithinBatch(inFlightBatch.firstOffset(), inFlightBatch.lastOffset())) {
+                    // For the case when batch.firstOffset < start offset <= batch.lastOffset, we will be having some
+                    // acquired records that need to move to archived state despite their delivery count.
+                    inFlightBatch.maybeInitializeOffsetStateUpdate();
+                }
+
+                // Case when the state of complete batch is valid
+                if (inFlightBatch.offsetState() == null) {
+                    releaseAcquisitionLockOnTimeoutForCompleteBatch(inFlightBatch, stateBatches, memberId);
+                } else { // Case when batch has a valid offset state map.
+                    releaseAcquisitionLockOnTimeoutForPerOffsetBatch(inFlightBatch, stateBatches, memberId, firstOffset, lastOffset);
+                }
+            }
+
+            if (!stateBatches.isEmpty() && !isWriteShareGroupStateSuccessful(stateBatches)) {
+                // Even if write share group state RPC call fails, we will still go ahead with the state transition.
+                log.error("Failed to write the share group state on acquisition lock timeout for share partition: {}-{} memberId {}. " +
+                                "Proceeding with state transition.", groupId, topicIdPartition, memberId);
+            }
+            // Update the cached state and start and end offsets after releasing the acquisition lock on timeout.
```

Comment:
And can we have blank lines before these // comments to make it easier to scan the code visually